### ISSUE TITLE
Error transforming S57 ENC to Postgres using ogr2ogr

### ISSUE THREAD

**Issue by chewmag** [OSGeo/gdal#4751]
## Expected behavior and actual behavior.
Hi, 
Transforming specific s57 ENC to Postgres using ogr2ogr results in error.

Expected behavior: transformation completes without errors
Actual behavior: error during transformation. The error is as follows: 

`ERROR 1: ERROR:  current transaction is aborted, commands ignored until end of transaction block
ERROR 1: ERROR:  current transaction is aborted, commands ignored until end of transaction block
ERROR 1: ERROR:  current transaction is aborted, commands ignored until end of transaction block
ERROR 1: ERROR:  current transaction is aborted, commands ignored until end of transaction block
ERROR 1: ERROR:  current transaction is aborted, commands ignored until end of transaction block
ERROR 1: CREATE TABLE "public"."bcnsaw" ( "ogc_fid" SERIAL, PRIMARY KEY ("ogc_fid"), "rcid" NUMERIC(10,0), "prim" NUMERIC(3,0), "grup" NUMERIC(3,0), "objl" NUMERIC(5,0), "rver" NUMERIC(3,0), "agen" NUMERIC(5,0), "fidn" NUMERIC(10,0), "fids" NUMERIC(5,0), "lnam" VARCHAR(16), "lnam_refs" varchar[], "ffpt_rind" INTEGER[], "bcnshp" INTEGER, "colour" varchar[], "colpat" varchar[], "condtn" INTEGER, "conrad" INTEGER, "convis" INTEGER, "datend" VARCHAR, "datsta" VARCHAR, "elevat" FLOAT8, "height" FLOAT8, "marsys" INTEGER, "natcon" varchar[], "nobjnm" VARCHAR, "objnam" VARCHAR, "perend" VARCHAR, "persta" VARCHAR, "status" varchar[], "veracc" FLOAT8, "verdat" INTEGER, "verlen" FLOAT8, "inform" VARCHAR, "ninfom" VARCHAR, "ntxtds" VARCHAR, "picrep" VARCHAR, "scamax" INTEGER, "scamin" INTEGER, "txtdsc" VARCHAR, "recdat" VARCHAR, "recind" VARCHAR, "sordat" VARCHAR, "sorind" VARCHAR, "wkb_geometry" geometry(POINT,1) )
ERROR:  current transaction is aborted, commands ignored until end of transaction block
ERROR 1: ERROR:  relation "bcnsaw" does not exist
ERROR 1: ERROR:  relation "bcnsaw" does not exist
ERROR 1: no COPY in progress

Warning 1: Attributes PEREND ignored, not in expected schema.
No more warnings will be issued for this dataset.
Start Time: 15:36:39,33
Finish Time: 15:36:42,57`  

## Steps to reproduce the problem.

Run ogr2ogr transformation from s57 to Postgres using the following command:
ogr2ogr -skipfailures -append -s_srs EPSG:4326 -t_srs EPSG:4326 -f PostgreSQL PG:"dbname='my_db' host='localhost' port='7738' user='postgres' password='my_password'" GB4X0001.000

Postgres version: 13.2 
ENC: [GB4X0001.000](https://iho.int/iho_pubs/standard/S-64/S-64_Edition_3.0.2/TDS/S-64_e3.0.2_ENC_Unencrypted_TDS.zip)
GB4X0001.000 file resides inside S-64_e3.0.2_ENC_Unencrypted_TDS\3.3 Settings\ENC_ROOT directory of S-64_e3.0.2_ENC_Unencrypted_TDS.zip archive.

## Operating system

Windows 10 PRO

## GDAL version and provenance

v3.3.3,  from gdal-303-1928-x64-core.msi
https://www.gisinternals.com/query.html?content=filelist&file=release-1928-x64-gdal-3-3-3-mapserver-7-6-4.zip

**Comment by jratike80** [comment]
Does it help to drop `-append`? Can you create that table by using pgAdmin or some other SQL client with
```
CREATE TABLE "public"."bcnsaw" ( "ogc_fid" SERIAL, PRIMARY KEY ("ogc_fid"), "rcid" NUMERIC(10,0), "prim" NUMERIC(3,0), "grup" NUMERIC(3,0), "objl" NUMERIC(5,0), "rver" NUMERIC(3,0), "agen" NUMERIC(5,0), "fidn" NUMERIC(10,0), "fids" NUMERIC(5,0), "lnam" VARCHAR(16), "lnam_refs" varchar[], "ffpt_rind" INTEGER[], "bcnshp" INTEGER, "colour" varchar[], "colpat" varchar[], "condtn" INTEGER, "conrad" INTEGER, "convis" INTEGER, "datend" VARCHAR, "datsta" VARCHAR, "elevat" FLOAT8, "height" FLOAT8, "marsys" INTEGER, "natcon" varchar[], "nobjnm" VARCHAR, "objnam" VARCHAR, "perend" VARCHAR, "persta" VARCHAR, "status" varchar[], "veracc" FLOAT8, "verdat" INTEGER, "verlen" FLOAT8, "inform" VARCHAR, "ninfom" VARCHAR, "ntxtds" VARCHAR, "picrep" VARCHAR, "scamax" INTEGER, "scamin" INTEGER, "txtdsc" VARCHAR, "recdat" VARCHAR, "recind" VARCHAR, "sordat" VARCHAR, "sorind" VARCHAR, "wkb_geometry" geometry(POINT,1) );
```

**Comment by rouault** [comment]
I believe the error messages are truncated. The most interesting part would be before the first " ERROR: current transaction is aborted, commands ignored until end of transaction block"

**Comment by chewmag** [comment]
> Does it help to drop `-append`? Can you create that table by using pgAdmin or some other SQL client with
> 
> ```
> CREATE TABLE "public"."bcnsaw" ( "ogc_fid" SERIAL, PRIMARY KEY ("ogc_fid"), "rcid" NUMERIC(10,0), "prim" NUMERIC(3,0), "grup" NUMERIC(3,0), "objl" NUMERIC(5,0), "rver" NUMERIC(3,0), "agen" NUMERIC(5,0), "fidn" NUMERIC(10,0), "fids" NUMERIC(5,0), "lnam" VARCHAR(16), "lnam_refs" varchar[], "ffpt_rind" INTEGER[], "bcnshp" INTEGER, "colour" varchar[], "colpat" varchar[], "condtn" INTEGER, "conrad" INTEGER, "convis" INTEGER, "datend" VARCHAR, "datsta" VARCHAR, "elevat" FLOAT8, "height" FLOAT8, "marsys" INTEGER, "natcon" varchar[], "nobjnm" VARCHAR, "objnam" VARCHAR, "perend" VARCHAR, "persta" VARCHAR, "status" varchar[], "veracc" FLOAT8, "verdat" INTEGER, "verlen" FLOAT8, "inform" VARCHAR, "ninfom" VARCHAR, "ntxtds" VARCHAR, "picrep" VARCHAR, "scamax" INTEGER, "scamin" INTEGER, "txtdsc" VARCHAR, "recdat" VARCHAR, "recind" VARCHAR, "sordat" VARCHAR, "sorind" VARCHAR, "wkb_geometry" geometry(POINT,1) );
> ```

No, removing the append option does not help. DDL statement "create table public.bcnsaw" is perfectly correct and creates a specified table when executed alone.

**Comment by rouault** [comment]
> s perfectly correct and creates a specified table when executed alone.

you should look if there are error messages *before* the first " ERROR: current transaction is aborted, commands ignored until end of transaction block"

**Comment by chewmag** [comment]
Unfortunately, there are no errors before the first "transaction is aborted error" from Postgres. Running ogr2ogr in debug mode produces the following logs:
[gdal_debug.log](https://github.com/OSGeo/gdal/files/7492029/gdal_debug.log)
Some data from ENC is eventually written into Postgres though.

**Comment by rouault** [comment]
I managed to reproduce.
Fix in https://github.com/OSGeo/gdal/pull/4961
You can add -oo RECODE_BY_DSSI=YES to ask the S57 driver to recode from LATIN1 to UTF-8

**Comment by chewmag** [comment]
> I managed to reproduce. Fix in #4961 You can add -oo RECODE_BY_DSSI=YES to ask the S57 driver to recode from LATIN1 to UTF-8

@rouault Thanks for the feedback. Setting RECODE_BY_DSSI=YES inside OGR_S57_OPTIONS fixes the problem.
